jira.go

```diff
@@ -48,6 +49,11 @@ func NewClient(httpClient *http.Client, baseURL string) (*Client, error) {
 		httpClient = http.DefaultClient
 	}
 
+	// ensure the baseURL contains a trailing slash so that all paths are preserved in later calls
+	if !strings.HasSuffix(baseURL, "/") {
```

Comment:
Maybe this is just me, but I think I'd rather return an error.  I'm a little hesitant to assume intent.  Maybe some upstream library is stripping is or ruining the data passed in.  I think it'd be more helpful to explain the exact problem and where it occurred.

Reply:
@rbriski it does not produce an error at the time of initialization. When a request is made, the jira instance returns a 404 resource not found because the final url constructed from `NewRequest` can be wrong if the trailing `/` is omitted. This PR is help users not make the same mistake we did.